README.md

```diff
@@ -231,10 +231,59 @@ iex> PersonTranscoder.encode!(%{name: "Jane Doe",
 
 This can be used to, for example, build Ecto changesets via a `changeset/2` functions and to render HAL responses to HTTP requests.
 
+#### Patching
+
+ExHal.Transcoder also supports modifying objects with [JSON patch](https://tools.ietf.org/html/rfc6902).
+
+```elixir
+defmodule PetTranscoder do
+  use ExHal.Transcoder
+
+  defproperty "name"
+  defproperty "animalType",   param: species, protected: true
+  deflink     "favoriteToy",  param: :favorite_toy
+  deflinks    "friends"
+end
+```
+
+Create an object from the transcoder, then modify it
+
+```elixir
+iex> spot = PetTranscoder.decode!(doc)
+%{name: "Spot",
+  species: "dog",
+  favorite_toy: "http://a.co/56guxwO"
+  friends: ["https://petbook.com/u/fifi", "https://petbook.com/u/fido"]}
```

Comment:
In this case the example is a decoded object, not a HAL document.  Like in this example:
```elixir
  test "Object representation test" do
    defmodule AnotherPersonTranscoder do
      use ExHal.Transcoder

      defproperty "name"
      defproperty "mailingAddress", param: :address
      deflink     "app:department", param: :department_url
      deflinks    "tags"
    end

    json = ~s(
      {
        "name": "Jane Doe",
        "mailingAddress": "123 Main St",
        "_links": {
          "app:department": { "href": "http://example.com/dept/42" },
          "tags": [
            {"href": "foo:1"},
            {"href": "http://2"},
            {"href": "urn:1"}
          ]
        }
      }
    )

    json
    |> ExHal.Document.parse!
    |> AnotherPersonTranscoder.decode!
    |> IO.inspect
end
=> %{address: "123 Main St", department_url: "http://example.com/dept/42",
  name: "Jane Doe", tags: ["foo:1", "http://2", "urn:1"]}
```

The pet example could be made into a HAL doc by doing something like this:

```elixir
spot |> PetTranscoder.patch!(json_patches) |> PetTranscoder.encode!
```